src/GafferScene/Rename.cpp

```diff
@@ -0,0 +1,760 @@
+{
+	return
+		input == filterPlug() ||
+		input == inPlug()->childNamesPlug() ||
+		input == namePlug()
+	;
+}
+
+void Rename::hashNameMap( const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	FilteredSceneProcessor::hash( nameMapPlug(), context, h );
+
+	if( !(filterValue( context ) & PathMatcher::DescendantMatch) )
+	{
+		return;
+	}
+
+	ConstInternedStringVectorDataPtr inputChildNames = inPlug()->childNamesPlug()->getValue();
+	ScenePlug::PathScope childScope( context );
+	ScenePlug::ScenePath childPath = context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName );
+	childPath.push_back( InternedString() ); // Room for child name
+
+	bool haveRenames = false;
+	IECore::MurmurHash renamesHash;
+	for( const auto &childName : inputChildNames->readable() )
+	{
+		childPath.back() = childName;
+		childScope.setPath( &childPath );
+		if( filterValue( childScope.context() ) & PathMatcher::ExactMatch )
+		{
+			haveRenames = true;
+			namePlug()->hash( renamesHash );
+		}
+		renamesHash.append( childName );
+	}
+
+	if( !haveRenames )
+	{
+		return;
+	}
+
+	h.append( renamesHash );
+}
+
+IECore::ConstObjectPtr Rename::computeNameMap( const Gaffer::Context *context ) const
+{
+	if( !(filterValue( context ) & PathMatcher::DescendantMatch) )
+	{
+		// No renaming here.
+		return g_emptyNameMap;
+	}
+
+	// Children are possibly being renamed.
+
+	vector<pair<InternedString, InternedString>> renames;
+	std::unordered_set<InternedString> usedNames;
+
+	ConstInternedStringVectorDataPtr inputChildNames = inPlug()->childNamesPlug()->getValue();
+
+	ScenePlug::PathScope childScope( context );
+	ScenePlug::ScenePath inputChildPath = context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName );
+	inputChildPath.push_back( InternedString() ); // Room for child name
+
+	for( const auto &inputChildName : inputChildNames->readable() )
+	{
+		inputChildPath.back() = inputChildName;
+		childScope.setPath( &inputChildPath );
+		if( filterValue( childScope.context() ) & PathMatcher::ExactMatch )
+		{
+			const string outputChildName = namePlug()->getValue();
+			if( outputChildName.size() && outputChildName != inputChildName.string() )
+			{
+				renames.push_back( { inputChildName, outputChildName } );
+				continue;
+			}
+		}
+		// This child not being renamed.
+		usedNames.insert( inputChildName );
+	}
+
+	if( !renames.size() )
+	{
+		return g_emptyNameMap;
+	}
+
+	// We've renamed some things, but we need to make sure the new
+	// names are unique with respect to everything that wasn't renamed.
+
+	boost::format namePrefixSuffixFormatter( "%s%d" );
+
+	NameMapDataPtr nameMapData = new NameMapData;
+	for( auto &[inputName, outputName] : renames )
+	{
+		if( usedNames.find( outputName ) != usedNames.end() )
+		{
+			// Uniqueify the name
+			string prefix;
+			int suffix = IECore::StringAlgo::numericSuffix( outputName.string(), 1, &prefix );
+
+			do
+			{
+				outputName = boost::str( namePrefixSuffixFormatter % prefix % suffix );
```

Comment:
I measured - `to_string` was close to 3x faster, so I've used that in https://github.com/GafferHQ/gaffer/pull/4842/commits/3bd5829f0af24a3a11b666828f0ce2d021e8d982 (also updated ChildNamesMap since that's where I copied the formatter from). Note though that we have bigger problems if we have enough name clashes for this to be significant - the strategy here is quadratic time and assumes there won't be many clashes.

As an aside, `fmtlib` (and eventually `std::format()`) are _so_ much nicer (and faster) than `boost::format()` - I'm tempted to start shifting everything over to that.

Reply:
Ooh, `fmt::format( "{}{}", prefix, suffix )` knocks almost 40% off `to_string()`. I'll hold off on doing that until we've decided what we're doing with the formatting for the output (if we're using fmtlib, I might want to add it to the dependencies more officially).